api/base/serializers.py

```diff
@@ -13,6 +12,27 @@
 from api.base.exceptions import InvalidQueryStringError, Conflict
 
 
+def format_relationship_links(related_link, self_link, rel_meta, self_meta):
+
+    ret = {
+        'links': {
+            'related': {
+                'href': related_link,
+                'meta': rel_meta
+            },
+            'self': {
+                'href': self_link,
+                'meta': self_meta
+            }
+        }
+    }
+
+    # Not currently returning self links
+    if not ret['links']['self']['href']:
```

Comment:
I believe the plan was to not include the relationship if it's not valid or doesn't exist. There was a lot of discussion around this, but it's a valid way to go and fairly easy to check for in a variety of languages. I don't recall if `parent: null` is valid from a spec standpoint, so the latter is safer in that case.